### DISCUSSION TITLE
Delayed sources on reconcile

### DISCUSSION THREAD

**Discussion by wibed** [fluxcd/flux2#2824] [Q&A] [answered]
i update my source files

either template files, value file or related. and cant run 
`flux reconcile whatever` afterwards, without the need to wait for flux
to do its magic.

why does reconcile not use latest changes and what do i have to do to manually replay whatever flux is doing 
implementing those changes?

**Comment by stefanprodan** [comment, marked as answer]
There is a `flux reconcile kustomization --with-source` and `flux reconcile helmrelease --with-source`, this will tell source-controller to pull the latest revision from Git, then tells kustomize-controller or helm-controller to reconcile with the new revision.

**Reply by stefanprodan** [reply]
A better option is to just do `flux reconcile source git`, and the reconciliation of all things that use that source will happen in the background.
